tests/devices/qutrit_mixed/test_qutrit_mixed_measure.py

```diff
@@ -0,0 +1,507 @@
+        mp1 = qml.state()
+        assert get_measurement_function(mp1, state=1) is reduce_density_matrix
+
+    def test_prod_trace_method(self):
+        """Test that the expectation value of a product uses the trace method."""
+        prod = qml.prod(*(qml.GellMann(i, 1) for i in range(8)))
+        assert get_measurement_function(qml.expval(prod), state=1) is trace_method
+
+    def test_hermitian_trace(self):
+        """Test that the expectation value of a hermitian uses the trace method."""
+        mp = qml.expval(qml.THermitian(np.eye(3), wires=0))
+        assert get_measurement_function(mp, state=1) is trace_method
+
+    def test_hamiltonian_sum_of_terms(self):
+        """Check that the sum of terms method is used when Hamiltonian."""
+        H = qml.Hamiltonian([2], [qml.GellMann(0, 1)])
+        state = qml.numpy.zeros((3, 3))
+        assert get_measurement_function(qml.expval(H), state) is sum_of_terms_method
+
+    def test_sum_sum_of_terms(self):
+        """Check that the sum of terms method is used when sum of terms"""
+        S = qml.prod(*(qml.GellMann(i, 1) for i in range(8))) + qml.prod(
+            *(qml.GellMann(i, 2) for i in range(8))
+        )
+        state = qml.numpy.zeros((3, 3))
+        assert get_measurement_function(qml.expval(S), state) is sum_of_terms_method
+
+    def test_probs_compute_probabilities(self):
+        """Check that compute probabilities method is used when probs"""
+        assert get_measurement_function(qml.probs(), state=1) is calculate_probability
+
+    def test_var_compute_variance(self):
+        """Check that the compute variance method is used when variance"""
+        obs = qml.GellMann(0, 1)
+        assert get_measurement_function(qml.var(obs), state=1) is calculate_variance
+
+
+@pytest.mark.parametrize(
+    "obs",
+    [
+        qml.GellMann(2, 2),
+        qml.GellMann(1, 8),
+        qml.GellMann(0, 5),
+        (qml.GellMann(0, 5) @ qml.GellMann(1, 8)),
+    ],
+)
+@pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+class TestApplyObservableEinsum:
+    """Tests that observables are applied correctly for trace method."""
+
+    num_qutrits = 3
+    dims = (3**num_qutrits, 3**num_qutrits)
+
+    def test_apply_observable_einsum(self, obs, three_qutrit_state, ml_framework):
+        """Tests that unbatched observables are applied correctly to a unbatched state."""
+        res = apply_observable_einsum(
+            obs, qml.math.asarray(three_qutrit_state, like=ml_framework), is_state_batched=False
+        )
+        expected = get_expanded_op_mult_state(obs, three_qutrit_state)
+        expected = expected.reshape([3] * (3 * 2))
+
+        assert qml.math.get_interface(res) == ml_framework
+        assert qml.math.allclose(res, expected)
+
+    def test_apply_observable_einsum_batched(self, obs, three_qutrit_batched_state, ml_framework):
+        """Tests that unbatched observables are applied correctly to a batched state."""
```

Comment:
Does it make sense to test the other two cases (batched observable/unbatched state, and batched/batched)?

Reply:
Maybe, but I am not sure if observables are supposed to be batched I don't think they are for `default.qubit`. Currently the code won't work for batched observables, but it could be added by adding `...` in the `apply_observable_einsum` implementation.

Reply:
Okay if there is not batching support for the qubit case let's leave it out to make things consistent here, then.